docs/aws/audit/elbmonitoring/rules/classic_elb_security_group.mdx

```diff
@@ -23,6 +23,208 @@ CBP
+
+4. In the bottom panel, click on the "Security" tab. Here, you can see the security groups associated with the selected ELB.
+
+5. Check if there is at least one valid security group associated with the ELB. If there are no security groups or if the associated security groups are invalid, then it is a misconfiguration.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access the Elastic Load Balancer (ELB) resources.
+
+2. Once the AWS CLI is set up, you can list all the load balancers in your account using the following command:
+
+   ```
+   aws elbv2 describe-load-balancers
+   ```
+
+   This command will return a JSON output with details about all the load balancers in your account.
+
+3. To check the security groups associated with each load balancer, you can use the following command:
+
+   ```
+   aws elbv2 describe-load-balancer-attributes --load-balancer-arn <load-balancer-arn>
+   ```
+
+   Replace `<load-balancer-arn>` with the ARN of the load balancer you want to check. This command will return a JSON output with the attributes of the specified load balancer, including the associated security groups.
+
+4. To check if the security group is valid, you can use the following command:
+
+   ```
+   aws ec2 describe-security-groups --group-ids <security-group-id>
+   ```
+
+   Replace `<security-group-id>` with the ID of the security group you want to check. This command will return a JSON output with the details of the specified security group. If the security group does not exist or is not associated with the load balancer, the command will return an error.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary AWS SDK for Python (Boto3) if you haven't done so already. You can do this by running the command `pip install boto3` in your terminal.
+
+2. Import the necessary modules and establish a session with AWS using your credentials. Here's a basic setup:
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='us-west-2'  # or any other region you prefer
+)
+```
+
+3. Now, create a client for ELB (Elastic Load Balancer) and retrieve all the load balancers:
+
+```python
+elb = session.client('elbv2')
+load_balancers = elb.describe_load_balancers()
+```
+
+4. Iterate over each load balancer and check if it has at least one security group:
+
+```python
+for lb in load_balancers['LoadBalancers']:
+    if not lb['SecurityGroups']:
+        print(f"ELB {lb['LoadBalancerName']} does not have any security groups.")
+```
+
+This script will print the names of all load balancers that do not have any security groups. Please replace `'YOUR_ACCESS_KEY'` and `'YOUR_SECRET_KEY'` with your actual AWS access key and secret key.
+</Accordion>
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>